### PR TITLE
Eslint

### PR DESCRIPTION
**Pull request by trust2065**
Use `eslint` and `pre-commit` to lint before commit.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "env": {
+    "browser": true,
+    "es6": true
+  },
+  "extends": "eslint:recommended",
+  "globals": {
+    "Atomics": "readonly",
+    "SharedArrayBuffer": "readonly"
+  },
```
Comment (josephj) [inline .eslintrc.json]: I think you don't have these 2 global variables. Remove it?

**Comment by josephj** [comment]
It's nicer if you can provide `.nvmrc` to specify the node version.

image

File: src/index.js [hunk 2]
```diff
@@ -1,39 +1,40 @@
...
+        updateAnimation(`section${index}`, "remove");
+        updateAnimation(`section${index + 1}`, "add");
+      } else if (direction === "up") {
+        updateAnimation(`section${index - 1}`, "add");
+        updateAnimation(`section${index}`, "remove");
+      }
     }
-  }
-});
+  });
```
Comment (josephj) [inline src/index.js]: Why do you need this anonymous function block?

File: package.json [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
-  "name": "Choco-Li",
+  "name": "choco_li",
   "email": "trust2065@gmail.com",
   "license": "UNLICENSED",
+  "pre-commit": [
+    "lint"
+  ],
   "scripts": {
     "start": "set NODE_ENV=production&&webpack --mode=development",
     "dev": "set NODE_ENV=development&&webpack --mode=development --hot",
```
[the remark this answers is not in the document]
Comment (trust2065) [reply package.json]: This will get fixed in other PR.

**Pull request merged**